lib/pub_cache.dart

```diff
@@ -52,10 +49,30 @@ class PubCache {
     return dir.existsSync() ? dir.listSync() : [];
   }
 
+  /**
+   * Return applications that have been installed via `pub global activate`.
+   */
   List<Application> getGlobalApplications() => _applications;
 
+  /**
+   * Get all the packages and their versions that have been installed into the
+   * cache.
+   */
   List<PackageRef> getPackageRefs() => _packageRefs;
 
+  /**
+   * Return the list of package names (not versions) that area available in the
+   * cache.
+   */
+  List<String> getCachedPackages() =>
+      new Set.from(getPackageRefs().map((p) => p.name)).toList();
```

Comment:
Yeah, I don't know about the call frequency; likely infrequently? I like the fact that the current implementation has less room for errors.